src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,108 @@
+#include "EndpointQueueFilter.h"
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+#include <string_view>
+
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+
+namespace {
+
+bool IsValidMdnsHostName(uint8_t * hostName)
+{
+    for (size_t i = 0; i < sizeof(hostName); ++i)
+    {
+        char ch_data = hostName[i];
+        if (!((ch_data >= '0' && ch_data <= '9') || (ch_data >= 'A' && ch_data <= 'F')))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool IsMdnsBroadcastPacket(const void * endpoint, const IPPacketInfo & pktInfo, const System::PacketBufferHandle & pktPayload)
+{
+    // if the packet is not a broadcast packet to mDNS port, drop it.
+    VerifyOrReturnValue(pktInfo.DestPort == 5353, false);
+#if INET_CONFIG_ENABLE_IPV4
+    ip_addr_t mdnsIPv4BroadcastAddr = IPADDR4_INIT_BYTES(224, 0, 0, 251);
+    VerifyOrReturnValue(pktInfo.DestAddress == Inet::IPAddress(mdnsIPv4BroadcastAddr), false);
```

Comment:
If IPV4 is enabled, this will always not check if the packet is IPv6, and immeidately return false.

Please layer the changes to avoid VerifyOrReturnValue. This is not a `verify` case.

This should be:

```
if (pktInfo.DestAddress == Inet::IPAddress(mdnsIPv4BroadcastAddr)
{
    return true;
}
```

Reply:
done